### ISSUE TITLE
test list is always created

### ISSUE THREAD

**Issue by LuisBeton** [doxygen/doxygen#7702]
With following option in V1.8.18 from configuration file

```
GENERATE_TESTLIST = NO
```
I expect to see no Test File entry in tree view inside index.html and also no generated test.html.
With V1.1.18 the test list option (set to no) is not working for me and it is treated like it is set to yes.

I have a lot of parsed source files (*.c) with \test inside but I don't want to use test file feature.
If I remove every parsed \test then no test list stuff is generated (maybe due to no found stuff then option GENERATE_TESTLIST must not be considered).

This works great with V1.8.17 and same configuration file.

**Comment by fettox** [comment]
It seems like GENERATE_TODOLIST and GENERATE_BUGLIST are also affected.
Am I missing something here?

**Comment by albert-github** [comment]
- check with `doxygen -x` what your settings are.
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by LuisBeton** [comment]
Try mentioned options in attached example project: [doxy_github_issue_7702.zip](https://github.com/doxygen/doxygen/files/4476649/doxy_github_issue_7702.zip).

**Comment by albert-github** [comment]
Looks like that during some refactoring some code disappeared in utils.cpp:
- 1.8.17
```
void addRefItem(const std::vector<ListItemInfo> &sli,
    const char *key,
    const char *prefix, const char *name,const char *title,const char *args,Definition *scope)
{
  //printf("addRefItem(sli=%p,key=%s,prefix=%s,name=%s,title=%s,args=%s)\n",sli,key,prefix,name,title,args);
  if (key && key[0]!='@') // check for @ to skip anonymous stuff (see bug427012)
  {
    for (const ListItemInfo &lii : sli)
    {
      RefList *refList = Doxygen::xrefLists->find(lii.type);
      if (refList
          &&
          (
           // either not a built-in list or the list is enabled
           (lii.type!="todo"       || Config_getBool(GENERATE_TODOLIST)) &&
           (lii.type!="test"       || Config_getBool(GENERATE_TESTLIST)) &&
           (lii.type!="bug"        || Config_getBool(GENERATE_BUGLIST))  &&
           (lii.type!="deprecated" || Config_getBool(GENERATE_DEPRECATEDLIST))
          )
         )
      {
        RefItem *item = refList->getRefItem(lii.itemId);
        ASSERT(item!=0);

        item->prefix = prefix;
        item->scope  = scope;
        item->name   = name;
        item->title  = title;
        item->args   = args;

        refList->insertIntoList(key,item);

      }
    }
  }
}
```
- 1.8.18
```
void addRefItem(const std::vector<RefItem*> &sli,
    const char *key,
    const char *prefix, const char *name,const char *title,const char *args,const Definition *scope)
{
  //printf("addRefItem(sli=%d,key=%s,prefix=%s,name=%s,title=%s,args=%s)\n",(int)sli.size(),key,prefix,name,title,args);
  if (key && key[0]!='@') // check for @ to skip anonymous stuff (see bug427012)
  {
    for (RefItem *item : sli)
    {
        item->setPrefix(prefix);
        item->setScope(scope);
        item->setName(name);
        item->setTitle(title);
        item->setArgs(args);
        item->setGroup(key);
    }
  }
}
```

**Comment by doxygen** [comment]
The idea was to move the filtering to `DocXRefItem::parse()` but this is not enough it seems.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
